{
  try {
    resource.close();
  }
 catch (  IOException e) {
  }
}
